src/linter.ts

```diff
@@ -131,19 +131,40 @@ class Linter {
             throw new Error(`formatter '${formatterName}' not found`);
         }
 
-        const output = formatter.format(this.failures, this.fixes);
+        // Sort failures by failure's line and character.
+        const failures = this.sortFailures(this.failures);
```

Comment:
I'd suggest to move the sorting of failures to each formatter that needs it. Formatters for machine consumption shouldn't rely on any order.
That means there are 4 formatters affected: `prose`, `verbose`, `stylish` and `code-frame`.